Add powerpc64le-unknown-linux-musl support

Gelbpunkt opened the pull request:
~~Still trying to figure out how to get this to be tested in CI.~~ Was able to test the Docker image locally, works fine.

<details>
<summary>Cross-compiled using the Docker image, ran on native target:</summary>

```
aludra:~$ ./rustup-init

Welcome to Rust!

This will download and install the official compiler for the Rust
programming language, and its package manager, Cargo.

Rustup metadata and toolchains will be installed into the Rustup
home directory, located at:

  /home/adrian/.rustup

This can be modified with the RUSTUP_HOME environment variable.

The Cargo home directory is located at:

  /home/adrian/.cargo

This can be modified with the CARGO_HOME environment variable.

The cargo, rustc, rustup and other commands will be added to
Cargo's bin directory, located at:

  /home/adrian/.cargo/bin

This path will then be added to your PATH environment variable by
modifying the profile file located at:

  /home/adrian/.profile

You can uninstall at any time with rustup self uninstall and
these changes will be reverted.

Current installation options:


   default host triple: powerpc64le-unknown-linux-musl
     default toolchain: stable (default)
               profile: default
  modify PATH variable: yes

1) Proceed with standard installation (default - just press enter)
2) Customize installation
3) Cancel installation
>1

info: profile set to 'default'
info: default host triple is powerpc64le-unknown-linux-musl
info: syncing channel updates for 'stable-powerpc64le-unknown-linux-musl'
info: latest update on 2025-04-03, rust version 1.86.0 (05f9846f8 2025-03-31)
info: downloading component 'cargo'
  9.4 MiB /   9.4 MiB (100 %)   2.6 MiB/s in  2s
info: downloading component 'clippy'
  3.2 MiB /   3.2 MiB (100 %)   2.6 MiB/s in  1s
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
 25.6 MiB /  25.6 MiB (100 %)   4.8 MiB/s in  6s
info: downloading component 'rustc'
 99.1 MiB /  99.1 MiB (100 %)   6.7 MiB/s in 19s
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
 21.2 MiB /  21.2 MiB (100 %)   6.3 MiB/s in  3s
info: installing component 'rust-std'
 25.6 MiB /  25.6 MiB (100 %)  11.4 MiB/s in  2s
info: installing component 'rustc'
 99.1 MiB /  99.1 MiB (100 %)  11.5 MiB/s in  8s
info: installing component 'rustfmt'
info: default toolchain set to 'stable-powerpc64le-unknown-linux-musl'

  stable-powerpc64le-unknown-linux-musl installed - rustc 1.86.0 (05f9846f8 2025-03-31)


Rust is installed now. Great!

To get started you may need to restart your current shell.
This would reload your PATH environment variable to include
Cargo's bin directory ($HOME/.cargo/bin).

To configure your current shell, you need to source
the corresponding env file under $HOME/.cargo.

This is usually done by running one of the following (note the leading DOT):
. "$HOME/.cargo/env"            # For sh/bash/zsh/ash/dash/pdksh
source "$HOME/.cargo/env.fish"  # For fish
source $"($nu.home-path)/.cargo/env.nu"  # For nushell
```

</details>

Successful runs:
https://github.com/rust-lang/rustup/actions/runs/14528297588/job/40763746894
https://github.com/rust-lang/rustup/actions/runs/14528297588/job/40763744397

Gelbpunkt committed:
Add powerpc64le-unknown-linux-musl support

Signed-off-by: Jens Reidel <adrian@travitia.xyz>

rami3l reviewed (approved):
@Gelbpunkt Thanks for the PR, and LGTM!

PS: Before merging I'd still like to see if it works on CI because we are in the middle of a new release right now (sorry :'/) To test `# skip-pr skip-master` targets you may simply add another tiny commit on top that removes `skip-pr`. When you are confident with out changes you can simply drop it.

Gelbpunkt commented:
> @Gelbpunkt Thanks for the PR, and LGTM!
> 
> PS: Before merging I'd still like to see if it works on CI because we are in the middle of a new release right now (sorry :'/) To test `# skip-pr skip-master` targets you may simply add another tiny commit on top that removes `skip-pr`. When you are confident with out changes you can simply drop it.

Sure thing :) I added a commit on top and it built fine, should I add a revert commit or drop it and force push?

rami3l commented:
> Sure thing :) I added a commit on top and it built fine, should I add a revert commit or drop it and force push?

@Gelbpunkt Please go ahead :D

Gelbpunkt commented:
> > Sure thing :) I added a commit on top and it built fine, should I add a revert commit or drop it and force push?
> 
> @Gelbpunkt Please go ahead :D

I dropped the commit and force pushed, hope that was okay.
